pkgs/build-support/node/build-npm-package/hooks/npm-config-hook.sh

```diff
@@ -0,0 +1,76 @@
+npmSetupHook() {
+    echo "Executing npmSetupHook"
+
+    runHook preConfigure
+
+    echo "Configuring npm"
+
+    export HOME=$TMPDIR
+    export npm_config_nodedir="@nodeSrc@"
+
+    npm config set cache "$npmDeps"
+    npm config set offline true
+    npm config set progress false
+
+    local cacheLockfile="$npmDeps/package-lock.json"
+    local srcLockfile="$PWD/package-lock.json"
+
+    echo "Validating consistency between $srcLockfile and $cacheLockfile"
+
+    if ! @diff@ "$srcLockfile" "$cacheLockfile"; then
+      # If the diff failed, first double-check that the file exists, so we can
+      # give a friendlier error msg.
```

Comment:
Line wrapping manually here looks a bit weird since lines below are way longer
```suggestion
      # If the diff failed, double-check that the file exists so we can give a friendlier error message.
```